File: teste.m
%% Teste da função GOLDEN
clear
clc

%% Inicializa variáveis
chute = 3;

%% Executa a função
[phi, iter] = golden(chute);

%% Imprime o resultado
str = [num2str(iter, '%u'), ' iterações: ϕ = ', num2str(phi, '%1.15f')];
disp(str)

File: teste.c
// Cálculo da razão áurea (phi)
#include <stdio.h>
#include <stdlib.h>
#include "golden.h"

int main() {

		double phi = 3;
		unsigned int iter = 0;

		golden(&phi, &iter);

		printf("%u iterações: ϕ = %1.15f\n",iter,phi);
		return EXIT_SUCCESS;

}
